src/ascent/runtimes/flow_filters/ascent_runtime_relay_filters.cpp

```diff
@@ -1240,6 +1243,161 @@ RelayIOLoad::execute()
+:Filter()
+{
+// empty
+}
+
+//-----------------------------------------------------------------------------
+BlueprintFlatten::~BlueprintFlatten()
+{
+// empty
+}
+
+//-----------------------------------------------------------------------------
+void
+BlueprintFlatten::declare_interface(Node &i)
+{
+    i["type_name"]   = "false";
+    i["port_names"].append() = "in";
+    i["output_port"] = "false";
+}
+
+//-----------------------------------------------------------------------------
+bool
+BlueprintFlatten::verify_params(const conduit::Node &params,
+                           conduit::Node &info)
+{
+    info.reset();
+    bool res = true;
+
+    if(! params.has_child("path") ||
+       ! params["path"].dtype().is_string() )
+    {
+        info["errors"].append() = "Missing required string parameter 'path'";
+    }
+
+    return res;
+    //return verify_io_params(params,info);
+}
+
+
+//-----------------------------------------------------------------------------
+void
+BlueprintFlatten::execute()
+{
+    std::string path, protocol;
+    path = params()["path"].as_string();
+    path = output_dir(path);
+
+    if(params().has_child("protocol"))
+    {
+        protocol = params()["protocol"].as_string();
+    }
+
+    if(!input("in").check_type<DataObject>())
+    {
+        // error
+        ASCENT_ERROR("Blueprint flatten requires a DataObject input");
+    }
+
+    DataObject *data_object  = input<DataObject>("in");
+    if(!data_object->is_valid())
+    {
+      return;
+    }
+    std::shared_ptr<Node> n_input = data_object->as_node();
+
+    Node *in = n_input.get();
+    Node selected;
+    if(params().has_path("fields"))
+    {
+      std::vector<std::string> field_selection;
+      const conduit::Node &flist = params()["fields"];
+      const int num_fields = flist.number_of_children();
+      if(num_fields == 0)
+      {
+        ASCENT_ERROR("Blueprint flatten field selection list must be non-empty");
+      }
+      for(int i = 0; i < num_fields; ++i)
+      {
+        const conduit::Node &f = flist.child(i);
+        if(!f.dtype().is_string())
+        {
+           ASCENT_ERROR("Blueprint flatten field selection list values must be a string");
+        }
+        field_selection.push_back(f.as_string());
+      }
+      detail::filter_fields(*in, selected, field_selection, graph());
+    }
+    else
+    {
+      // select all fields
+      selected.set_external(*in);
+    }
+    Node output;
+#ifdef ASCENT_MPI_ENABLED
+    MPI_Comm mpi_comm = MPI_Comm_f2c(flow::Workspace::default_mpi_comm());
+    blueprint::mpi::mesh::flatten(selected,
+		                  params(),
+				  output,
+  				  mpi_comm);
+#else
+    blueprint::mesh::flatten(selected,
+		             params(),
+			     output);
+
+#endif
+
+    std::string result_path;
+    int rank = 0;
+    int root = 0;
+ 
+#ifdef ASCENT_MPI_ENABLED
+    MPI_Comm_rank(mpi_comm, &rank);
+#endif
+
+    if(rank == root)
+    {
+        if(protocol.empty())
```

Comment:
we can simplify this here, here is a suggestion:

if empty - use csv, otherwise just pass the protocol to `relay::io::save()`

for the non csv case, we want to support `hdf5`, `json`, `yaml` etc -- passing the protocol simplifies plumbing.

Reply:
Sounds good. I'll fix my cases.